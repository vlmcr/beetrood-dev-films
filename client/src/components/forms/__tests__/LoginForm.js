import React from 'react';
import LoginForm from '../LoginForm';
import ReactDOM from 'react-dom';
import { MemoryRouter } from "react-router-dom";
import { toHaveAttribute, toHaveTextContent} from "@testing-library/jest-dom/matchers"

import { queries } from "@testing-library/dom";

expect.extend({toHaveAttribute, toHaveTextContent})

test("LoginForm should render correct", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <MemoryRouter>
      <LoginForm />
    </MemoryRouter>,
    div
  );

  // expect(div.querySelector("#email").type).toBe("email")
  // expect(div.querySelector("#email")).toHaveAttribute("type", "email");
  // expect(div.querySelector("label")).toHaveTextContent("Email");

  const emailEl = queries.getByLabelText(div, /email/i)
  expect(emailEl).toHaveAttribute("type", "email");
})