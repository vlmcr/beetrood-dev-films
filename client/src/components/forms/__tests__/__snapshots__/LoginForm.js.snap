// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should render correct 1`] = `
<form
  class="ui form"
>
  <div
    class="field"
  >
    <label
      for="email"
    >
      email
    </label>
    <input
      id="email"
      name="email"
      placeholder="Email"
      type="email"
      value=""
    />
    <div
      style="color: rgb(154, 63, 56);"
    />
  </div>
  <div
    class="field"
  >
    <label
      for="password"
    >
      Password
    </label>
    <input
      id="password"
      name="password"
      placeholder="password"
      type="text"
      value=""
    />
    <div
      style="color: rgb(154, 63, 56);"
    />
  </div>
  <div
    class="ui fluid buttons"
  >
    <button
      class="ui button primary"
      data-test-id="login-button"
    >
      Login
    </button>
    <div
      class="or"
    />
    <a
      class="ui button"
      href="/"
    >
      Cancel
    </a>
  </div>
</form>
`;
